### PR TITLE
Add github action to push data changes to the gh-pages branch

### LINKED ISSUE #6: Add a Github Pages API to dapi-management to export the Merkle trees with signatures
Replaces this [issue](https://github.com/api3dao/tasks/issues/198)

### PR DESCRIPTION
**Pull request by pieterdub**
Closes #6 

`github-pages.yml` has been created to include new changes to the `data` folder into `gh-pages` branch. Which has been set up to deploy `gh-pages` to the github pages api. 

Steps include:

1.  Checkout main
2. Switch branch to `gh-pages`
3. checkout data folder on main into `gh-pages`
4. Commit and push if there are any changes

To access github pages one can navigate to:

`https://api3dao.github.io/dapi-management/data/<merkle-tree-path>/current-hash.json`

For example 
`https://api3dao.github.io/dapi-management/data/dapi-fallback-merkle-tree-root/current-hash.json`

## Current Github api page after this branch is merged it should automatically deploy github pages with the correct timestamp
image

### CODE REVIEW

File: .github/workflows/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# GitHub Actions Workflow: Deploy to GitHub Pages
+
+This GitHub Actions workflow automates the process of deploying new files from the main branch to the gh-pages branch, updating the contents of a `data` folder.
```
Comment (mcoetzee) [inline .github/workflows/README.md:3]: Hmm this doesn't sound quite right to me. 
```
automates the process of deploying new files
```
How I understand it, is that the GH workflow detects changes to the `data` folder on the `main` branch, and pushes those changes to the `gh-pages` branch. That's it. GH pages then takes care of deploying the `gh-pages` branch.

**Review by mcoetzee — CHANGES_REQUESTED** [review]
Nice one 👍

I just had a look at the `gh-pages` branch, and can we re-create it, so that we can get rid of all your test commits.

File: .github/workflows/github-pages.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    - name: Commit and push if there are changes
+      run: |
+        if [ -n "$(git status --porcelain)" ]; then
+          echo "Changes detected in gh-pages branch."
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git add .
+          git commit -m "Update gh-pages with new files"
+          git push --set-upstream origin gh-pages
```
[the remark this answers is not in the document]
Comment (pieterdub) [reply .github/workflows/github-pages.yml]: Yes you are correct

**Review by mcoetzee — APPROVED** [review]
Nice one 👍

**Pull request merged**